src/browser/browser.zig

```diff
@@ -57,30 +59,44 @@ pub const user_agent = "Lightpanda/1.0";
 // A browser contains only one session.
 // TODO allow multiple sessions per browser.
 pub const Browser = struct {
-    session: Session = undefined,
-    agent: []const u8 = user_agent,
+    loop: *Loop,
+    session: ?*Session,
+    allocator: Allocator,
+    session_pool: SessionPool,
 
-    const uri = "about:blank";
+    const SessionPool = std.heap.MemoryPool(Session);
```

Comment:
Is it OK to use the heap directly here instead of an allocator passed by the caller?

Reply:
If you mean that it's in `std.heap`, it's the same as `std.heap.ArenaAllocator`. It's a specialized ArenaAllocator that, knowing the type, can keep a free-list available. There's a bunch of places I want to add it (i.e. the Loop creating all those Completions)

It's initialized in browser's `init` as:

```zig
    pub fn init(allocator: Allocator, loop: *Loop) Browser {
        return .{
             // ...
            .session_pool = SessionPool.init(allocator),
        };
    }
```

So it does use the passed-in allocator.

https://www.openmymind.net/Zig-MemoryPool-Allocator/